cng/sha.go

```diff
@@ -104,7 +104,8 @@ type shaXHash struct {
 
 func newSHAX(id string, key []byte) *shaXHash {
 	var flag bcrypt.AlgorithmProviderFlags
-	if len(key) > 0 {
+	if key != nil {
```

Comment:
It seems to me that it's unusual to treat an empty slice differently from `nil` slice... this is what comes to mind first for me, instead:

```go
func newSHAXHMAC(id string, key []byte) *shaXHash {
	return newSHAXCore(id, bcrypt.ALG_HANDLE_HMAC_FLAG, key)
}
func newSHAX(id string) *shaXHash {
	return newSHAXCore(id, 0, nil)
}
func newSHAXCore(id string, flag bcrypt.AlgorithmProviderFlags, key []byte) *shaXHash {
	h, err := loadSha(id, flag)

```

Maybe since it's not an exported func it doesn't matter much.

Reply:
I like the proposal. The implementation is more robust and clearer.